setup-env

```diff
@@ -4,53 +4,201 @@ if [ -z "$ZSH_NAME"] && [ "x$0" = "x./setup-env" ]; then
+           usage
+           clean_up
+           return 0
+           ;;
+        --)
+           shift
+           break
+           ;;
+    esac
+done
+
+if [ "$(whoami)" = "root" ]; then
+    echo "ERROR: do not use the BSP as root. Exiting..."
+fi
+
+if [ ! -e $1/conf/local.conf ]; then
+    build_dir_setup_enabled="true"
 else
-    mkdir -p "$BDIR"
-    BDIR=`readlink -f "$BDIR"`
+    build_dir_setup_enabled="false"
+fi
+
+if [ "$build_dir_setup_enabled" = "true" ] && [ -z "$MACHINE" ]; then
+    usage
+    echo -e "ERROR: You must set MACHINE when creating a new build directory."
+    clean_up
+    return 1
 fi
+
+if [ -z "$SDKMACHINE" ]; then
+    SDKMACHINE='x86_64'
+fi
+
+if [ "$build_dir_setup_enabled" = "true" ] && [ -z "$DISTRO" ]; then
+    usage
+    echo -e "ERROR: You must set DISTRO when creating a new build directory."
+    clean_up
+    return 1
+fi
+
 # Allow for multiple layer sets,
 # default just "layers"
 if [ "x$BMETA" = "x" ]; then
     BMETA="layers"
 fi
-OEROOT=`readlink -f "$OEROOT/$BMETA"`
+OEROOT=`readlink -f "$CWD/$BMETA"`
 export OEROOT
-export BITBAKEDIR="$OEROOT/bitbake"
-. $OEROOT/scripts/oe-buildenv-internal && \
-    $OEROOT/scripts/oe-setup-builddir && \
-    [ -n "$BUILDDIR" ] && cd $BUILDDIR
-TEMPLATECONF=`cat $BUILDDIR/conf/templateconf.cfg`
+
+export TEMPLATECONF=${TEMPLATECONF:-$OEROOT/meta-testdistro/conf}
+
+. $OEROOT/openembedded-core/oe-init-build-env $CWD/$1 > /dev/null
+
+# if conf/local.conf not generated, no need to go further
+if [ ! -e conf/local.conf ]; then
+    clean_up && return 1
+fi
+
+generated_config=
+if [ "$build_dir_setup_enabled" = "true" ]; then
+
+    # Change settings according environment
+    sed -e "s,MACHINE ??=.*,MACHINE ??= '$MACHINE',g" \
+        -e "s,SDKMACHINE ?=.*,SDKMACHINE ?= '$SDKMACHINE',g" \
+        -e "s,DISTRO ?=.*,DISTRO ?= '$DISTRO',g" \
+        -e "s,PACKAGE_CLASSES ?=.*,PACKAGE_CLASSES ?= '$PACKAGE_CLASSES',g" \
+        -i conf/local.conf
+
+    for s in $HOME/.oe $HOME/.yocto; do
+        if [ -e $s/site.conf ]; then
+            echo "Linking $s/site.conf to conf/site.conf"
+            ln -s $s/site.conf conf
+        fi
+    done
+
+    generated_config=1
+fi
+
+cat <<EOF
+Welcome to the meta-tegra test distribution.
+
+The Yocto Project has extensive documentation about OE including a
+reference manual which can be found at:
+    http://yoctoproject.org/documentation
+For more information about OpenEmbedded see their website:
+    http://www.openembedded.org/
+
+You can now run 'bitbake <target>'
+Common targets are:
+    core-image-minimal
+    meta-toolchain
+    meta-toolchain-sdk
+    adt-installer
+    meta-ide-support
+EOF
+
+if [ -n "$generated_config" ]; then
+    cat <<EOF
+Your build environment has been configured with:
+    MACHINE=$MACHINE
+    SDKMACHINE=$SDKMACHINE
+    DISTRO=$DISTRO
+EOF
+else
+    echo "Your configuration files at $1 have not been touched."
+fi
+
+TDROOT=`readlink -f "$CWD/$BMETA"`
+export TDROOT
```

Comment:
What's this for?

Reply:
This was exported in the original version of the your `setup-env` script, so I assumed you were using it. If not, I'll take it out.

Reply:
Although, now that I'm making some other changes here, I'm realizing that OEROOT gets unset by the `oe-init-build-env`, but I need this path later on. So even if it doesn't need to exported, I'm going move this up so I can use it later on.